src/QMCDrivers/DMC/WalkerControl.h

```diff
@@ -166,7 +193,15 @@ class WalkerControl : public MPIObjectBase
   const IndexType num_ranks_;
   ///0 is default
   IndexType SwapMode;
-  ///any temporary data includes many ridiculous conversions of integral types to and from fp
+  /** large amount of state data MPI_AllReduce once per step size is LE_MAX + number of ranks.
+   *  partial enum to access elements is the enum above dumped into WalkerControl's namespace
+   *  includes many integral types converted to and from fp
+   *  \todo MPI in the 21st centure allows user defined data types that would allow all this to benefit from actual
+   *        C++ type safety and still can have performant collective operation efficiency.
```

Comment:
MPI is C only. I don't think it knows C++ type safety. The derived type in MPI is intended for performance, not for safety. It is not useful in this particular case.

Reply:
I looks like its intended so everyone doesn't do things like this constantly.